cranelift/codegen/src/isa/s390x/inst/mod.rs

```diff
@@ -68,6 +68,23 @@ fn inst_size_test() {
     assert_eq!(32, std::mem::size_of::<Inst>());
 }
 
+/// A register pair. Enum so it can be destructured in ISLE.
```

Comment:
Again the question, why have another definition here when it is already defined in ISLE?  I think it should be one place or the other, not both.

Reply:
It's a little unfortunate to have the redundancy here, but this is currently the best way to have the constructors from an enum available when the declaration is in rust.

Reply:
I guess my question is, why *does* the declaration have to be in rust?  Why not just have it in ISLE, and use the auto-generated definition from Rust code, just like with the other enums defined in ISLE?

Reply:
Unfortunately the reason in this case is that it allows us to derive `Copy` for `RegPair` and `WritableRegPair`. Without deriving `Copy` the register pair values get moved into the instructions that use them, and borrow errors are generated for any subsequent use of those values. This might be a pattern that we should enable in ISLE, but for now it's easiest to define those types in rust instead.

Reply:
Ah, I see.  Thanks for the explanation.